ui/selectors/gator-permissions/gator-permissions.ts

```diff
@@ -354,6 +361,170 @@ export const getTokenTransferPermissionsByOrigin = createSelector(
+  (gatorPermissionsMap): Map<string, number> => {
+    const sitePermissionCounts = new Map<string, number>();
+
+    Object.values(gatorPermissionsMap).forEach((permissionTypeMap) => {
+      Object.values(permissionTypeMap).forEach((permissions) => {
+        permissions.forEach(
+          (
+            permission: StoredGatorPermissionSanitized<
+              Signer,
+              PermissionTypesWithCustom
+            >,
+          ) => {
+            if (permission?.siteOrigin) {
+              const currentCount =
+                sitePermissionCounts.get(permission.siteOrigin) || 0;
+              sitePermissionCounts.set(permission.siteOrigin, currentCount + 1);
+            }
+          },
+        );
+      });
+    });
+
+    return sitePermissionCounts;
+  },
+);
+
+/**
+ * Get the total count of unique sites (combining traditional connections and gator permissions).
+ *
+ * @param state - The current state
+ * @returns Total number of unique sites
+ * @example
+ * const totalSites = getTotalUniqueSitesCount(state);
+ *
+ * // 5 (sites with connections or gator permissions, excluding snaps)
+ */
+export const getTotalUniqueSitesCount = createSelector(
+  [
+    getConnectedSitesListWithNetworkInfo,
+    getUniqueSiteOriginsFromTokenTransferPermissions,
+  ],
+  (sitesConnectionsList, gatorPermissionSiteOrigins): number => {
+    // Get unique site origins from site connections (excluding snaps)
+    const connectedSiteOrigins = Object.keys(sitesConnectionsList).filter(
+      (site) => !isSnapId(site),
+    );
+
+    // Combine both lists and get unique sites
+    const allUniqueSites = new Set([
+      ...connectedSiteOrigins,
+      ...gatorPermissionSiteOrigins,
+    ]);
+
+    return allUniqueSites.size;
+  },
+);
+
+/**
+ * Get merged connections list that includes sites with both traditional connections
+ * and gator permissions. Sites with only gator permissions will have minimal connection data.
+ *
+ * @param state - The current state
+ * @returns Merged connections list object
+ * @example
+ * const connections = getMergedConnectionsListWithGatorPermissions(state);
+ *
+ * // {
+ * //   'https://example.com': {
+ * //     addresses: ['0x123'],
+ * //     origin: 'https://example.com',
+ * //     name: 'Example Site',
+ * //     advancedPermissionsCount: 3,
+ * //     ...
+ * //   },
+ * //   'https://gator-only.com': {
+ * //     addresses: [],
+ * //     origin: 'https://gator-only.com',
+ * //     name: 'Gator Only Site',
+ * //     advancedPermissionsCount: 2,
+ * //     ...
+ * //   }
+ * // }
+ */
+export const getMergedConnectionsListWithGatorPermissions = createSelector(
+  [
+    getConnectedSitesListWithNetworkInfo,
+    getGatorPermissionCountsBySiteOrigin,
+    (state: AppState) => state,
+  ],
+  (sitesConnectionsList, gatorPermissionCounts, state) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const mergedConnections: Record<string, any> = { ...sitesConnectionsList };
```

Comment:
Instead of using any, how about defining a Connection interface here for better type safety and readability?
```ts
interface Connection {
  addresses: string[];
  origin: string;
  name: string;
  iconUrl: string | null;
  subjectType: SubjectType;
  networkIconUrl: string;
  networkName: string;
  extensionId: string | null;
  advancedPermissionsCount?: number;
}
```